src/main/java/net/fabricmc/loom/configuration/providers/mappings/utils/URLFileSpec.java

```diff
@@ -35,18 +33,20 @@
 
 import net.fabricmc.loom.api.mappings.layered.MappingContext;
 import net.fabricmc.loom.api.mappings.layered.spec.FileSpec;
-import net.fabricmc.loom.util.DownloadUtil;
+import net.fabricmc.loom.util.download.DownloadException;
 
-public record URLFileSpec(URL url) implements FileSpec {
+public record URLFileSpec(String url) implements FileSpec {
```

Comment:
URL.hashCode [makes a network request](https://stackoverflow.com/questions/2348399/why-does-java-net-urls-hashcode-resolve-the-host-to-an-ip).

I think DLBuilder should also take a URL 👍

Reply:
> URL.hashCode [makes a network request](https://stackoverflow.com/questions/2348399/why-does-java-net-urls-hashcode-resolve-the-host-to-an-ip).

Precisely my point, the workaround isn't needed anymore since it's a String now.